src/utils.rs

```diff
@@ -20,25 +72,72 @@ pub(crate) fn to_snake_case(input: &str) -> String {
     output
 }
 
-pub fn generate_struct_code(table_name: &str, rows: &Vec<TableColumn>) -> String {
+pub fn generate_enum_code(
+    enum_name: &str,
+    enum_rows: &Vec<UserDefinedEnums>,
+    enable_serde: bool,
+) -> String {
+    let rs_enum_name = to_pascal_case(enum_name);
+    let mut enum_code = String::new();
+
+    let serde_derives = if enable_serde {
+        ", serde::Serialize, serde::Deserialize"
+    } else {
+        ""
+    };
+    enum_code.push_str(&format!(
+        "#[derive(sqlx::Type, Debug, Clone, Eq, PartialEq{})]\n",
+        serde_derives
+    ));
+    enum_code.push_str(&format!(r#"#[sqlx(type_name = "{}")]"#, enum_name));
+    enum_code.push_str("\n");
+    enum_code.push_str(&format!("pub enum {} {{\n", rs_enum_name));
+
+    for row in enum_rows.iter().filter(|e| &e.enum_name == enum_name) {
+        enum_code.push_str(&format!(r#"  #[sqlx(rename = "{}")]"#, row.enum_value));
+        enum_code.push_str("\n");
+        enum_code.push_str(&format!("  {},\n", to_pascal_case(&row.enum_value)))
+    }
+
+    enum_code.push_str("}\n");
+
+    enum_code
+}
+
+pub fn generate_struct_code(
+    table_name: &str,
+    rows: &Vec<TableColumn>,
+    enable_serde: bool,
+) -> String {
     let struct_name = to_pascal_case(table_name);
     let mut struct_code = String::new();
 
+    let serde_derives = if enable_serde {
+        ", serde::Serialize, serde::Deserialize"
+    } else {
+        ""
+    };
```

Comment:
I actually wonder whether instead of hardcoding this for serde, it's worth making more extensible by just allowing people to provide extra headers (sort of like how tonic build works for gRPC)

Reply:
That's a great idea. We could have new args like `--struct-derive` and `--enum-derive` that could take multiple values